### PR TITLE
Add support for Network Security Groups

### PR DESCRIPTION
**Pull request by vlappenbusch**

### CODE REVIEW

File: rpc/cloudagent/network/networksecuritygroup/moc_cloudagent_networksecuritygroup.proto
Comment (sbgms) [inline rpc/cloudagent/network/networksecuritygroup/moc_cloudagent_networksecuritygroup.proto:19]: >//TODO: Do we allow Both as an option? and add separate inbound/outbound rules? [](http://example.com/codeflow?start=15&length=79)

Remove comment. 
Both is not supported.

File: rpc/cloudagent/network/networksecuritygroup/moc_cloudagent_networksecuritygroup.proto
Comment (msftpgayam) [inline rpc/cloudagent/network/networksecuritygroup/moc_cloudagent_networksecuritygroup.proto:38]: Why do we have `sourceAddressPrefix` and `sourceAddressPrefixes`? similarly `destinationAddressPrefixes`, `sourcePortRanges` and `destinationPortRanges`?

File: rpc/common/moc_common_common.proto
Comment (msftpgayam) [inline rpc/common/moc_common_common.proto:321]: How come we don't have Delete?

File: rpc/cloudagent/network/networksecuritygroup/moc_cloudagent_networksecuritygroup.proto
Comment (vlappenbusch) [reply rpc/cloudagent/network/networksecuritygroup/moc_cloudagent_networksecuritygroup.proto:38]: This was set up to match the API of the azure NSGs that we're trying to match, but we're not using the repeated ones in practice - because sourceAddressPrefix or sourcePortRange can be a comma separated list of address prefixes or port ranges, we just reuse that when needed. The repeated versions can be removed if needed

File: rpc/common/moc_common_common.proto
Comment (vlappenbusch) [reply rpc/common/moc_common_common.proto:321]: I was following the operations of other SDN options in this list, I didn't see a delete for other SDN options. I can add Delete if needed for this list, but I didn't see it in many other places

File: rpc/cloudagent/network/networksecuritygroup/moc_cloudagent_networksecuritygroup.proto
Comment (msftpgayam) [reply rpc/cloudagent/network/networksecuritygroup/moc_cloudagent_networksecuritygroup.proto:38]: We need to discuss and decide whether we need the sourceAddressPrefixes or not.

File: rpc/common/moc_common_common.proto
Comment (msftpgayam) [reply rpc/common/moc_common_common.proto:321]: @apluchik could you check if this is correct? Don't we need `Delete`?

File: rpc/common/moc_common_networkcommon.proto
Comment (msftpgayam) [inline rpc/common/moc_common_networkcommon.proto:22]: Could you verify if these are being used somewhere?

**Review by msftpgayam — DISMISSED** [review]

**Comment by msftpgayam** [comment]
Let's not merge this PR until all the PRs are approved.

**Review by sbgms — APPROVED** [review]

**Review by sbgms — APPROVED** [review]

**Review by msftpgayam — APPROVED** [review]

**Pull request merged**
